Architecture.md

```diff
@@ -55,10 +61,62 @@
 
 ### Database architecture
 
-***TODO: [CLIENT] describe the database architecture, as referenced in the MCD in the README***
+As the database is unified and serves as a focal point of the application, everything is contained in the same logical database.
+The architecture is as follows:
+
+```mermaid
+erDiagram
+    USERS {
+        uuid id PK
+        varchar(40) username UK
+        varchar(100) email UK
+        varchar(255) password
+        bool is_admin "Default false"
+        uuid email_nonce UK "Nullable, default uuid()"
+        uuid password_nonce UK "Nullable"
+        timestamp created_at
+        timestamp updated_at "Nullable"
+    }
+    SSH_KEYS {
+        uuid id PK
+        text value "public SSH key content"
+        varchar(30) name
+        timestamp created_at
+        timestamp updated_at "Nullable"
+        uuid user_id FK
+    }
+    PROJECTS {
+        uuid id PK
+        varchar(40) name
+        jsonb config
+        timestamp created_at
+        timestamp updated_at "Nullable"
+        uuid user_id FK
+    }
+
+    USERS ||--o{ PROJECTS : manage
+    USERS ||--o{ SSH_KEYS : possess
+```
+
+As you can notice, every table has a uuid field as a primary key. Compared to `SERIAL` that is often used to identify a table, a uuid guarantees a better uniqueness across the database. A [Serial](https://www.postgresql.org/docs/current/datatype-numeric.html) may take up less space (4 bytes) than an [uuid](https://www.postgresql.org/docs/current/datatype-uuid.html) (16 bytes) but since it is a series of incrementing integers it offers information about the time of creation and makes it easier to guess the id whereas an uuid is generated randomly and nearly impossible to duplicate.
```

Comment:
```suggestion
As you can notice, every table has a uuid field as a primary key. Compared to `SERIAL` that is often used to identify a table, a uuid guarantees uniqueness across the whole database (and most likely across the whole universe). A [Serial](https://www.postgresql.org/docs/current/datatype-numeric.html) may take up less space (4 bytes) than a [uuid](https://www.postgresql.org/docs/current/datatype-uuid.html) (16 bytes) but since it is a series of incrementing integers it offers information about the time of creation and makes it easier to guess the id whereas a uuid is generated randomly and is nearly impossible to duplicate.
```